### DISCUSSION TITLE
How to write unit test for dropdown menu component with Vitest and React-testing-library

### DISCUSSION THREAD

**Discussion by lazarus2019** [radix-ui/primitives#2012] [Help]
How to write unit test for dropdown menu component with Vitest and React-testing-library, I can not get the dropdown content to expect it to be in the document

**Comment by Miloshinjo** [comment]
I recently wrote a test suite for a radix dropdown with Vitest and RTL.

Here is one test as example. 

```
test('displays the dropdown on trigger click', async () => {
    const user = userEvent.setup();
    const props = getProps();

    render(<DropdownMenu {...props} />);

    // Get trigger button and click it
    const triggerButton = screen.getByRole('button');
    await user.click(triggerButton);

    // Get menu and menu items
    const menu = screen.getByRole('menu');
    const menuItems = screen.getAllByRole('menuitem');

    // Assertions
    expect(triggerButton).toHaveAttribute('aria-expanded', 'true');
    expect(menu).toBeInTheDocument();
    expect(menuItems.length).toBe(props.items.length);
  });
```

**Reply by manvydas-tbol** [reply]
@Miloshinjo What if there is more than 1 menu on the screen? How would you distinct between those?

**Reply by snelsi** [reply]
Use [testId](https://testing-library.com/docs/queries/bytestid/) or get the specific dropdown button [by text](https://testing-library.com/docs/queries/bytext).
